### PR TITLE
Fix bad parameter reads on rover

### LINKED ISSUE #10510: Rover: "scan:695: scan past end of eeprom" displayed in SITL
When rover is run is SITL the message, "scan:695: scan past end of eeprom" is displayed on the console.  I suspect this is being caused by some parameter conversion code that is using an incorrect parameter name.  I think it is innocuous but let's get rid of this message if possible.

scan-past-end-of-eeprom

### PR DESCRIPTION
**Pull request by WickedShell**
This resolves #10510. What was happening is the rangefinder initializer was running before the AP_Param `_storage` initializer got to run. This lead to a `_storage.size()` of 0. The net effect is that we are unable to set the default orientation of the rangefinders. This is currently only happening on rover, but it depends entirely upon compiler decisions, and could easily happen on the other vehicles (or with the cross compiled output and we are just missing it).

The simple fix is to never try and manipulate a parameter during the constructor, which is what's presented here.

The only question I have is if there is any risk on the call to `AP_Param::setup_object_defaults(this, var_info);` misbehaving because it needs to read storage for any reason. It seems fine as far as I can tell, but I thought I'd ask if someone had a chance to audit it further.

This fixes #10510

### CODE REVIEW

File: APMrover2/Rover.h [hunk 1]
```diff
@@ -174,7 +174,7 @@ class Rover : public AP_HAL::HAL::Callbacks {
     AP_Baro barometer;
     Compass compass;
     AP_InertialSensor ins;
-    RangeFinder rangefinder{serial_manager, ROTATION_NONE};
```
Comment (peterbarker) [inline APMrover2/Rover.h]: I shouldn't mention the serial_manager singleton, should I?

Comment (WickedShell) [reply APMrover2/Rover.h]: That's actually not problematic, we stash the pointer (which is valid) and don't use it till init() so no race problems there (unless you are calling init before all the initializers run)

**Comment by rmackay9** [comment]
This was on the list to backport to Rover-3.5 but the merge conflicts are a bit much so I'll leave it to Rover-3.6 unless there are strong objections.

**Comment by WickedShell** [comment]
@rmackay9 it means that the default orientation isn't necessarily what it should be, and whenever this gets fixed you could see a sudden change in parameter value. (I'm not sure what order the cross compilier is doing the inits in though, so it could be less of a problem (or more) on the rover builds). In fact it's not even guaranteed between firmware compilations.

**Pull request merged**